lib/wasi/src/syscalls/mod.rs

```diff
@@ -2657,18 +2657,6 @@ pub fn path_rename<M: MemorySize>(
         source_path.to_str().as_ref().unwrap(),
         true
     ));
-    if state
-        .fs
-        .get_inode_at_path(
-            inodes.deref_mut(),
-            new_fd,
-            target_path.to_str().as_ref().unwrap(),
-            true,
-        )
```

Comment:
I believe we added this check because just checking would create the inode if didn't exist.

Perhaps we can just remove the if/return and make it:

```suggestion
        fs.get_inode_at_path(
            inodes.deref_mut(),
            new_fd,
            target_path.to_str().as_ref().unwrap(),
            true,
        );
```

Reply:
I don't think it's necessary, I think the test was really added for a wrong reason (avoid overwrite of destination while it should allow it). But creating the node doesn't do any harm, so lets do that.